### PR TITLE
[release/7.0] Port workload fixes from 6.0

### PR DESCRIPTION
**Pull request by joeloff**
- The updated task now provides metadata that will allow us to split the VSDROP generation into two separate archives. One which only contains workload packs (needed to share DROPs across VS versions to support multi-targeting) and another that only contains the manifest installers and workload components.
- The ZIP archive names have been updated to disambiguate its contents and support automatic drop creation during staging.

Most of the changes are in `eng` due to the Arcade update to obtain the latest copy of the workload build tasks. Those changes are unrelated to the workload fix, but appears to be necessary.

### CODE REVIEW

**Review by lewing — COMMENTED** [review]
worload changes lgtm, someone else should look over the arcade related stuff

**Comment by carlossanlop** [comment]
@joeloff are these changes considered tell-mode or ask-mode? If it's the latter, we need M2 approval.

**Comment by lewing** [comment]
this is build infrastructure supporting workload insertion and is tell mode as discussed in tactics

**Comment by carlossanlop** [comment]
Ok thanks, @lewing.

There are a lot of failures. Are any of them related to this?

**Comment by lewing** [comment]
The failures aren't because of the workload stuff, it looks like it is the sdk bump causing problems.

@marcpopMSFT is this because the SDK bumped the 6.0 bumped the 6.0.x runtime?

File: src/workloads/workloads.csproj
Comment (hoyosjs) [inline src/workloads/workloads.csproj:134]: How come nested in a p directory?

Comment (hoyosjs) [reply src/workloads/workloads.csproj:134]: Is it to avoid path length issues?

**Review by hoyosjs — APPROVED** [review]

**Comment by carlossanlop** [comment]
> This can probably wait until servicing has gone out

Meaning this should wait until after the RC2 snap?

File: src/workloads/workloads.csproj
Comment (joeloff) [reply src/workloads/workloads.csproj:134]: yup, p => packs, c => components... We've introduced additional naming elements for multitargeting and if we ever do SxS preview feature bands, then the package IDs can get quite long so I'm just avoiding things early on. And we already include the `packs` and `component` parts in the final artifact name, but we need separate folders to zip things up insolation.

**Comment by joeloff** [comment]
> > This can probably wait until servicing has gone out
> 
> Meaning this should wait until after the RC2 snap?

That's fine, as long as we get this into RC2 after the snap otherwise we're going to have to go through all the manual work again for VS.

**Review by ViktorHofer — APPROVED** [review]

**Comment by carlossanlop** [comment]
There's a conflict. @joeloff can you please resolve it?

**Comment by joeloff** [comment]
> There's a conflict. @joeloff can you please resolve it?

Done. I picked the RC1 SDK since that's what `darc` pulled in as part of the Arcade update.

**Comment by carlossanlop** [comment]
> That's fine, as long as we get this into RC2 after the snap otherwise we're going to have to go through all the manual work again for VS.

Thanks @joeloff. I added the "no merge" label. Please remove it when this is ready to merge after the snap, and ping me.

**Review by lewing — APPROVED** [review]

**Comment by joeloff** [comment]
Verified a signed internal build and everything looks as expected.

**Review by hoyosjs — APPROVED** [review]

**Comment by lewing** [comment]
failures are not related

```
      MSBuild version 17.4.0-preview-22416-02+5d102ae37 for .NET
      /datadisks/disk1/work/B5AA0966/p/wasm-test-runner/WasmTestRunner.proj : error : Failed to resolve SDK 'Microsoft.Build.NoTargets/3.5.0'. Package restore was successful but a package with the ID of "Microsoft.Build.NoTargets" was not installed.
      /datadisks/disk1/work/B5AA0966/p/wasm-test-runner/WasmTestRunner.proj : error MSB4236: The SDK 'Microsoft.Build.NoTargets/3.5.0' specified could not be found.
      Test Harness Exitcode is : 1
      To run the test:
      > set CORE_ROOT=/datadisks/disk1/work/B5AA0966/p
      > /datadisks/disk1/work/B5AA0966/w/A7060920/e/JIT/HardwareIntrinsics/X86/Avx2/LoadAlignedVector256NonTemporal_ro/LoadAlignedVector256NonTemporal_ro.sh
```

**Comment by hoyosjs** [comment]
Failures are https://github.com/dotnet/runtime/issues/75391 and PinObj_neg doesn't seem to have an associated issue, but there's other runs with the same failure mode - the output doesn't get more descriptive than `Test Harness Exitcode is : 1` and there's not much to go by nonetheless... Merging this as tell mode for workloads work. cc: @jeffschwMSFT @marek-safar

**Pull request merged**
